install/kubernetes/cilium/files/nodeinit/startup.bash

```diff
@@ -22,11 +22,85 @@ fi
 {{- end }}
 
 {{- if .Values.nodeinit.reconfigureKubelet }}
-# GKE: Alter the kubelet configuration to run in CNI mode
-echo "Changing kubelet configuration to --network-plugin=cni --cni-bin-dir={{ .Values.cni.binPath }}"
-mkdir -p {{ .Values.cni.binPath }}
-sed -i "s:--network-plugin=kubenet:--network-plugin=cni\ --cni-bin-dir={{ .Values.cni.binPath }}:g" /etc/default/kubelet
-echo "Restarting kubelet..."
+# Check if we're running on a GKE containerd flavor.
+GKE_KUBERNETES_BIN_DIR="/home/kubernetes/bin"
+if [[ -f "${GKE_KUBERNETES_BIN_DIR}/gke" ]] && command -v containerd &>/dev/null; then
+  echo "GKE *_containerd flavor detected..."
+
+  # (GKE *_containerd) Upon node restarts, GKE's containerd images seem to reset
+  # the /etc directory and our changes to the kubelet and Cilium's CNI
+  # configuration are removed. This leaves room for containerd and its CNI to
+  # take over pods previously managed by Cilium, causing these pods to become
+  # unmanaged. We rely on the fact that /home/kubernetes/bin/kubelet is not
```

Comment:
What is the guarantee that the kubelet won't change? Where is that "fact"? I think these are important to document in case this changes in the future.

Reply:
Unfortunately I couldn't find any documentation mentioning this fact — this is strictly an "observation", not a documented fact. It's one of the weakness of this fix/workaround, but I guess we'll have to deal with that. I'll update the comment though. :+1:

Reply:
How will we notice if that assumption break? I'm guessing none of our tests will detect that because we don't restart nodes in CI?

Reply:
This is of course a concern and a shortcoming of this approach. Unfortunately I don't have a good answer to it other than having CI restart nodes. 😞